baybe/utils/validation.py

```diff
@@ -66,3 +75,51 @@ def validator(self: Any, attribute: Attribute, value: Any) -> None:
 
 non_inf_float = _make_restricted_float_validator(allow_nan=True, allow_inf=False)
 """Validator for non-infinite floats."""
+
+
+def get_transform_objects(
+    objects: Sequence[_T],
+    df: pd.DataFrame,
+    allow_missing: bool,
+    allow_extra: bool,
+) -> list[_T]:
+    """Extract the objects relevant for transforming a given dataframe.
+
+    The passed object are assumed to have corresponding columns in the given dataframe,
+    identified through their name attribute. The function returns the subset of objects
+    that have a corresponding column in the dataframe and thus provide the necessary
+    information for transforming the dataframe.
+
+    Args:
+        objects: A collection of objects to be considered for transformation (provided
+            they have a match in the given dataframe).
+        df: The dataframe to be searched for corresponding columns.
+        allow_missing: Flag controlling if objects are allowed to have no corresponding
+            columns in the dataframe.
+        allow_extra: Flag controlling if the dataframe is allowed to have columns
+            that have no corresponding objects.
+
+    Raises:
+        ValueError: If the given objects and dataframe are not compatible
+            under the specified values for the Boolean flags.
+
+    Returns:
+        The (subset of) objects that need to be considered for the transformation.
+    """
+    names = [p.name for p in objects]
```

Comment:
Do we guarantee that the `objects` all have a `name` attribute? Or should this be checked for?

Reply:
Glad that you ask 👍🏼 But yes, `mypy` keeps us save here --> see that the input type is bound to `Parameter` and `Target`. Otherwise, it would have complained, indeed